server/lib/chat/chat-service.test.ts

```diff
@@ -232,6 +231,62 @@ describe('chat/chat-service', () => {
     await chatService.joinChannel(testChannel.name, user1.id)
   }
 
+  let textMessage: FakeDbTextChannelMessage
+  const addMessageToChannelMock = asMockedFunction(addMessageToChannel)
+
+  /**
+   * A helper method that mocks everything needed to send a message to a channel.
+   */
+  function mockTextMessage(
+    user: SbUser,
+    channel: ChannelInfo,
+    processedMessageString: string,
+    mentions: SbUser[],
+  ) {
+    textMessage = {
+      msgId: 'MESSAGE_ID',
+      userId: user.id,
+      userName: user.name,
+      channelId: channel.id,
+      sent: new Date('2023-03-11T00:00:00.000Z'),
+      data: {
+        type: ServerChatMessageType.TextMessage,
+        text: processedMessageString,
+        mentions: mentions.map(m => m.id),
+      },
+    }
+    // NOTE(2Pac): The `joinUserToChannel` calls already mock the return value of this function,
+    // so we use the `mockResolvedValueOnce` method to bump the return value to the top of the
+    // call stack.
+    addMessageToChannelMock.mockResolvedValueOnce(textMessage)
+    asMockedFunction(findUsersByName).mockResolvedValue(new Map(mentions.map(m => [m.name, m])))
+  }
+
+  /**
+   * A helper method which sends a chat message from a `user` to a `channel`, and expects the
+   * `client` has not received it. This is used as a test to make sure the users not in a channel
+   * don't receive messages for that channel.
+   */
+  async function expectMessageWasNotReceived(
+    user: SbUser,
+    channel: ChannelInfo,
+    client: InspectableNydusClient,
+  ) {
+    mockTextMessage(user, channel, 'Hello World!', [])
+
+    await chatService.sendChatMessage(channel.id, user.id, textMessage.data.text)
+
+    expect(client.publish).not.toHaveBeenCalledWith(getChannelPath(channel.id), {
+      action: 'message2',
+      message: toTextMessageJson(textMessage),
+      user: {
```

Comment:
You've reproduced the thing I asked you to fix elsewhere

Reply:
Interesting :)